loopy/target/c/__init__.py

```diff
@@ -841,8 +841,11 @@ def get_function_declaration(
             # subkernel launches occur only as part of entrypoint kernels for now
             from loopy.schedule.tools import get_subkernel_arg_info
             skai = get_subkernel_arg_info(kernel, subkernel_name)
+            if (not self.target.single_subkernel_is_entrypoint
+                and len(codegen_state.callables_table.items()) > 1):
+                raise LoopyError(f"""Kernel {kernel.name} has more than one subkernel.""")
```

Comment:
```suggestion
            from loopy.schedule.tools import get_subkernel_arg_info
            from loopy.tools import get_subkernels
            skai = get_subkernel_arg_info(kernel, subkernel_name)
            if (not self.target.single_subkernel_is_entrypoint
                    and len(get_subkernels(kernel)) > 1):
                raise LoopyError(f"Kernel '{kernel.name}' has more than one subkernel,"
                                 f" not allowed in {self.target}.")
```

I think it needs a small change. Subkernels at this stage are referred to as the different kernels separated by a `global barrier`.